subprojects/core/src/main/java/org/gradle/internal/service/scopes/DefaultGradleUserHomeScopeServiceRegistry.java

```diff
@@ -117,25 +120,27 @@ public Optional<ServiceRegistry> getCurrentServices() {
     public void release(ServiceRegistry registry) {
         lock.lock();
         try {
-            for (Map.Entry<File, Services> entry : servicesForHomeDir.entrySet()) {
+            release(servicesForHomeDir.entrySet().stream()
+                .filter(entry -> entry.getValue().registry == registry && entry.getValue().count != ZERO)
+                .findFirst());
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private void release(Optional<Map.Entry<File, Services>> release) {
+        release.ifPresent(
```

Comment:
why is the latest code already considered outdated?

Forcing me to commit this of kind silly code.


```suggestion
        release.ifPresentOrElse(
```